src/Pipeline.cpp

```diff
@@ -933,6 +933,8 @@ void Pipeline::prepare_jit_call_arguments(RealizationArg &outputs, const Target
                                           bool is_bounds_inference, JITCallArgs &args_result) {
     user_assert(defined()) << "Can't realize an undefined Pipeline\n";
 
+    user_assert(outputs.size() == this->outputs().size()) << "Realization requests " << outputs.size() << " output(s) for pipeline producing " << this->outputs().size() << " result(s).\n";
```

Comment:
Can I 'nit' here and say that at first read this didn't sound like an error message to me? Maybe wording like 

    The realization requests %x output(s) but the pipeline produces %y result(s).

Reply:
Full output is:
```Error: (at /usr/local/google/home/zalman/src/halide/test/error/realization_with_too_many_outputs.cpp:17)
Realization requests 2 output(s) for pipeline producing 1 result(s).```

Reply:
Changed.